src/main/java/nextstep/qna/domain/Answers.java

```diff
@@ -0,0 +1,47 @@
+package nextstep.qna.domain;
+
+import nextstep.qna.CannotDeleteException;
+import nextstep.users.domain.NsUser;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class Answers {
+
+    private final List<Answer> answers = new ArrayList<>();
+
+    public List<DeleteHistory> delete(NsUser loginUser) throws CannotDeleteException {
+        deleteValidation(loginUser);
+        return deleteHistories(loginUser);
+    }
+
+    private void deleteValidation(NsUser loginUser) throws CannotDeleteException {
+        boolean cannotDeleteAnswers = answers.stream().anyMatch(answer -> !answer.isOwner(loginUser));
+        if (cannotDeleteAnswers) {
```

Comment:
```suggestion
        if (isCannotDeleteAnswers(loginUser)) {
```

`stream` 로직을 의미를 드러네는 네이밍과 함께 메소드로 도출해내는건 어떨까요?